ai/speech_optimization.py

```diff
@@ -3,162 +3,169 @@
+        address_info = addressing_regex.match(status.group(5))
+        if status_type is StatusType.ADDRESS_BY_ID_PLAIN:
+            return f"{base_message} Addressing: Drone #{address_info.group(1)}"
+        elif status_type is StatusType.ADDRESS_BY_ID_INFORMATIVE:
+            return f"{base_message} Addressing: Drone #{address_info.group(1)} :: {address_info.group(3)}"
+    else:
+        return None
 
 
 async def optimize_speech(message: discord.Message, message_copy):
+    '''
+    This function allows status codes to be transformed into human-readable versions.
+    "5890 :: 200" > "5890 :: Code 200 :: Affirmative"
 
-    # Strip attachments only if a drone is optimized via hc!tso.
-    if is_optimized(message.author):
-        message_copy.attachments = []
+    Drones can append additional information after a status code.
+    Optimized drones cannot, and will have their message deleted if attempted.
+    '''
 
-    # If the message is written by a drone with speech optimization, and the message is NOT a valid message, delete it.
+    # Do not attempt to optimize non-drones.
+    if not is_drone(message.author):
+        return False
 
-    acceptable_status_code_message = plain_status_code_regex.match(message.content)
-    informative_status_code_message = informative_status_code_regex.match(message.content)
+    # Skip edge cases
+    if should_not_optimize(message):
+        return False
 
-    is_status_code = acceptable_status_code_message and acceptable_status_code_message.group(1) == get_id(message.author.display_name)
-    is_acceptable = message.content in get_acceptable_messages(message.author, message.channel.name) or is_status_code
+    # Attempt to find a status code message.
+    status = status_code_regex.match(message_copy.content)
+    if status is None and is_optimized(message.author):
+        await message.delete()
+        return True
 
-    if is_optimized(message.author) and not is_acceptable and message.channel.name not in CHANNEL_BLACKLIST and message.channel.category.name not in CATEGORY_BLACKLIST:
-        LOGGER.info("Deleting inappropriate message by optimized drone.")
+    # Confirm the status starts with the drone's ID
+    if status.group(2) != get_id(message.author.display_name):
+        LOGGER.info("Status did not match drone ID.")
+        await message.delete()
+        return True
+
+    # Determine status type
+    status_type = get_status_type(status)
+
+    # Delete unauthorized messages
+    if is_optimized(message.author) and status_type in (StatusType.INFORMATIVE, StatusType.ADDRESS_BY_ID_INFORMATIVE):
+        LOGGER.info("Deleting unauthorized message from optimized HexDrone.")
         await message.delete()
         return True
-    elif (informative_status_code_message or is_status_code) and is_drone(message.author):
-        LOGGER.info("Optimizing speech code for drone.")
-        message_copy.content = await print_status_code(message)
-        return False
     else:
```

Comment:
this else is unnecessary and confusing if you only log in its block but continue logic outside of it